bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java

```diff
@@ -2397,50 +2397,56 @@ public int compare(File file1, File file2) {
-        } else {
-            shell.ledgerIdFormatter = LedgerIdFormatter.newLedgerIdFormatter(shell.bkConf);
-        }
-        if (LOG.isDebugEnabled()) {
-            LOG.debug("Using ledgerIdFormatter {}", shell.ledgerIdFormatter.getClass());
-        }
+    public static void main(String[] argv) {
+        int res = -1;
+        try {
+            BookieShell shell = new BookieShell();
 
-        // entry format
-        if (cmdLine.hasOption(ENTRY_FORMATTER_OPT)) {
-            String val = cmdLine.getOptionValue(ENTRY_FORMATTER_OPT);
-            shell.entryFormatter = EntryFormatter.newEntryFormatter(val, shell.bkConf);
-        } else {
-            shell.entryFormatter = EntryFormatter.newEntryFormatter(shell.bkConf);
-        }
-        if (LOG.isDebugEnabled()) {
-            LOG.debug("Using entry formatter {}", shell.entryFormatter.getClass());
-        }
+            // handle some common options for multiple cmds
+            Options opts = new Options();
+            opts.addOption(CONF_OPT, true, "configuration file");
+            opts.addOption(LEDGERID_FORMATTER_OPT, true, "format of ledgerId");
+            opts.addOption(ENTRY_FORMATTER_OPT, true, "format of entries");
+            BasicParser parser = new BasicParser();
+            CommandLine cmdLine = parser.parse(opts, argv, true);
+
+            // load configuration
+            CompositeConfiguration conf = new CompositeConfiguration();
+            if (cmdLine.hasOption(CONF_OPT)) {
+                String val = cmdLine.getOptionValue(CONF_OPT);
+                conf.addConfiguration(new PropertiesConfiguration(
+                        new File(val).toURI().toURL()));
+            }
+            shell.setConf(conf);
 
-        int res = shell.run(cmdLine.getArgs());
-        System.exit(res);
+            // ledgerid format
+            if (cmdLine.hasOption(LEDGERID_FORMATTER_OPT)) {
+                String val = cmdLine.getOptionValue(LEDGERID_FORMATTER_OPT);
+                shell.ledgerIdFormatter = LedgerIdFormatter.newLedgerIdFormatter(val, shell.bkConf);
+            } else {
+                shell.ledgerIdFormatter = LedgerIdFormatter.newLedgerIdFormatter(shell.bkConf);
+            }
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Using ledgerIdFormatter {}", shell.ledgerIdFormatter.getClass());
+            }
+
+            // entry format
+            if (cmdLine.hasOption(ENTRY_FORMATTER_OPT)) {
+                String val = cmdLine.getOptionValue(ENTRY_FORMATTER_OPT);
+                shell.entryFormatter = EntryFormatter.newEntryFormatter(val, shell.bkConf);
+            } else {
+                shell.entryFormatter = EntryFormatter.newEntryFormatter(shell.bkConf);
+            }
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Using entry formatter {}", shell.entryFormatter.getClass());
+            }
+
+            res = shell.run(cmdLine.getArgs());
+        } catch (Exception e) {
```

Comment:
```suggestion
        } catch (Throwable e) {
```

otherwise in case of unchecked exceptions the user won't get the stacktrace

Reply:
Addressed. PTAL